core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileSupplier.java

```diff
@@ -124,29 +126,31 @@ static ProfileFileSupplier aggregate(ProfileFileSupplier... suppliers) {
             @Override
             public ProfileFile get() {
                 boolean refreshAggregate = false;
+                List<ProfileFile> nextValues = new ArrayList<>(suppliers.length);
                 for (ProfileFileSupplier supplier : suppliers) {
-                    if (didSuppliedValueChange(supplier)) {
+                    if (didSuppliedValueChange(supplier, nextValues)) {
                         refreshAggregate = true;
                     }
                 }
 
                 if (refreshAggregate) {
-                    refreshCurrentAggregate();
+                    refreshCurrentAggregate(nextValues);
                 }
 
                 return  currentAggregateProfileFile.get();
             }
 
-            private boolean didSuppliedValueChange(Supplier<ProfileFile> supplier) {
+            private boolean didSuppliedValueChange(Supplier<ProfileFile> supplier, List<ProfileFile> nextValues) {
                 ProfileFile next = supplier.get();
+                nextValues.add(next);
                 ProfileFile current = currentValuesBySupplier.put(supplier, next);
 
                 return !Objects.equals(next, current);
             }
 
-            private void refreshCurrentAggregate() {
+            private void refreshCurrentAggregate(List<ProfileFile> nextValues) {
                 ProfileFile.Aggregator aggregator = ProfileFile.aggregator();
-                currentValuesBySupplier.values().forEach(aggregator::addFile);
```

Comment:
This won't necessarily return the values in order since it's backed by a Map.

Reply:
Could we switch it to a linked hash map?

Reply:
Yes. The thing is we lose the `Concurrency` from the current map, but we can synchronize manually. Would that be a preferred solution?